### PR TITLE
added route for admin #533

### LINKED ISSUE #533: - Add admin route

### PR DESCRIPTION
**Pull request by MatteoGuarnaccia5**
## Description

Added admin page to IMS which has 2 cards - for units and usage statuses

## Testing instructions

Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking

closes #533

### CODE REVIEW

**Review by joelvdavies — COMMENTED** [review]
Just minor suggestions.

**Review by joshuadkitenge — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -35,10 +35,12 @@ import ManufacturerLandingPage from './manufacturer/manufacturerLandingPage.comp
 import Items from './items/items.component';
 import ItemsLandingPage from './items/itemsLandingPage.component';
 import ConfigProvider from './configProvider.component';
+import AdminPage from './admin/admin.component';
 
 export const paths = {
   any: '*',
   root: '/',
+  admin: '/admin-ims',
```
Comment (joshuadkitenge) [inline src/App.tsx:43]:
```suggestion
  admin: '/admin-ims/*',
```

File: src/App.tsx [hunk 2]
```diff
@@ -47,6 +49,8 @@ export const paths = {
   catalogueItem: '/catalogue/item/:catalogue_item_id',
   items: '/catalogue/item/:catalogue_item_id/items',
   item: '/catalogue/item/:catalogue_item_id/items/:item_id',
+  units: '/admin-ims/units',
+  usageStatus: '/admin-ims/usage-status',
```
Comment (joshuadkitenge) [inline src/App.tsx:53]:
```suggestion
```

File: src/App.tsx [hunk 3]
```diff
@@ -75,6 +79,9 @@ const router = createBrowserRouter([
       { path: paths.any, Component: ViewTabs },
       { path: paths.root, Component: HomePage },
       { path: paths.homepage, Component: HomePage },
+      { path: paths.admin, Component: AdminPage },
+      { path: paths.units, Component: AdminPage },
+      { path: paths.usageStatus, Component: AdminPage },
```
Comment (joshuadkitenge) [inline src/App.tsx:84]:
```suggestion
      { path: paths.admin, Component: AdminPage },
```

**Review by joshuadkitenge — CHANGES_REQUESTED** [review]

File: src/admin/admin.component.tsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  const navigateToAdminFunction = useNavigateToAdminFunction();
+  const adminFunction = useAdminFunction();
+
+  const adminBreadCrumbs: BreadcrumbsInfo | undefined = adminFunction
+    ? {
+        trail: [['', adminFunction ?? 'admin']],
+        full_trail: true,
+      }
+    : undefined;
+
```
[the remark this answers is not in the document]
Comment (joshuadkitenge) [reply src/admin/admin.component.tsx:44]: can the breadcrumbs be "Units" and "Usage status". it is currently "units" and  "usage-status" 

`
adminBreadcrumbs = {['units']:'Units',['usage-status']:'Usage status'}`

File: src/admin/admin.component.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+                      width: '100%',
+                      display: 'flex',
+                      justifyContent: 'center',
+                      alignItems: 'center',
+                      minWidth: 0,
+                    }}
+                  >
+                    <Grid>
+                      <Grid position={'relative'}>
+                        <Typography>Usage Status</Typography>
```
Comment (joelvdavies) [inline src/admin/admin.component.tsx:159]:
```suggestion
                        <Typography>Usage Statuses</Typography>
```
Could the names be changed to `Usage statuses` and `usage-statuses` to match units please.

File: src/admin/admin.component.tsx [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+                  </CardContent>
+                </Card>
+              </Button>
+            </Grid>
+          </Grid>
+        </Grid>
+      )}
+
+      {adminPageName === 'units' && <Units />}
+      {adminPageName === 'usage-status' && <UsageStatus />}
```
Comment (joelvdavies) [inline src/admin/admin.component.tsx:171]:
```suggestion
      {adminPageName === 'usage-statuses' && <UsageStatuses />}
```

**Review by joelvdavies — CHANGES_REQUESTED** [review]

File: src/admin/usageStatus.component.tsx
Comment (joelvdavies) [inline src/admin/usageStatus.component.tsx:1]: Could you rename this file to usageStatuses as well please.

**Review by joshuadkitenge — CHANGES_REQUESTED** [review]
https://github.com/ral-facilities/inventory-management-system/assets/83226114/08b9d999-0aea-4d2f-80f4-9fb93f9dd3e0

the breadcrumbs alignment is off compared to the other pages

File: src/admin/admin.component.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (joshuadkitenge) [reply src/admin/admin.component.tsx]: can you use a dictionary to assign the user friendly names please 
`adminBreadcrumbsNames[adminPageName ?? '']`

**Review by joshuadkitenge — CHANGES_REQUESTED** [review]
image

add no results page please for an invalid url

File: src/admin/admin.component.tsx [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+              justifyContent: 'center',
+            }}
+          >
+            <Typography sx={{ fontWeight: 'bold', textAlign: 'center' }}>
+              No results found
+            </Typography>
+            <Typography sx={{ textAlign: 'center' }}>
+              The admin function you searched for does not exist. Please
+              navigate home by pressing the home button at the top left of your
+              screen.
```
Comment (joshuadkitenge) [inline src/admin/admin.component.tsx:180]:
```suggestion
The admin URL route you're trying to access doesn't exist. Please return to the homepage by clicking the home button  at the top left of your screen.
```

File: src/admin/admin.component.tsx [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+      adminPageName === '' ? null : adminPageName.replace('/', '');
+    return adminPageName;
+  }, [location.pathname]);
+};
+
+function AdminPage() {
+  const navigateToAdminFunction = useNavigateToAdminFunction();
+  const adminPageName = useGetAdminPageName();
+
+  const adminBreadCrumbsNames: { [key: string]: [string, string] } = {
```
Comment (joshuadkitenge) [inline src/admin/admin.component.tsx:46]:
```suggestion
  const adminBreadCrumbsTrails: { [key: string]: [string, string] } = {
```

Can this also be defined outside of the component function please

File: src/admin/admin.component.tsx [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+    ['units']: ['Units', 'Units'],
+    ['usage-statuses']: ['Usage statuses', 'Usage statuses'],
```
Comment (joshuadkitenge) [inline src/admin/admin.component.tsx:48]:
```suggestion
    ['units']: ['units', 'Units'],
    ['usage-statuses']: ['usage-statuses', 'Usage statuses'],
```

**Comment by joshuadkitenge** [comment]
image
the no result page appears on the admin home page

File: src/admin/admin.component.tsx [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+                </Card>
+              </Button>
+            </Grid>
+          </Grid>
+        </Grid>
+      )}
+
+      {adminPageName === 'units' && <Units />}
+      {adminPageName === 'usage-statuses' && <UsageStatuses />}
+      {adminPageName !== undefined &&
```
Comment (joshuadkitenge) [inline src/admin/admin.component.tsx:165]:
```suggestion
      {adminPageName !== null &&
```

Comment (joshuadkitenge) [reply src/admin/admin.component.tsx:165]: This should stop the no results message from appearing on the admin home page

**Review by joshuadkitenge — APPROVED** [review]
LGTM

**Review by joelvdavies — APPROVED** [review]

**Pull request merged**